src/DataStructures.jl

```diff
@@ -16,9 +16,7 @@ module DataStructures
                  union, intersect, symdiff, setdiff, issubset,
                  searchsortedfirst, searchsortedlast, in
 
-    using Compat
-    using Compat.InteractiveUtils # for methodswith
-    import Compat: lastindex, pushfirst!, popfirst!
+    using InteractiveUtils # for methodswith
```

Comment:
`using InteractiveUtils: methodswith`

Also the `dict_support.jl` function that uses it needs to be updated to look for `iterate` rather than `start`/ `done`/ `next`.
Whether as part of this PR or later idk.

Reply:
I think that changes should me made when changing to the new iteration protocol.

Reply:
(I note that this is used for a very weird thing also, perhaps just a method error is better but w/e).

Reply:
Could do, thought it is actually not directly related to them.
Changing to new iteration protocol. is about changing types we define to that protocol.
Changing `dict_supports.jl` is about changing the types we *consume*  to be the new protocol.
But yes it can go in that PR,